src/bindings/js/node/scripts/download_runtime.js

```diff
@@ -0,0 +1,271 @@
+        `Directory '${destinationPath}' already exists. Skipping runtime downloading because 'ignoreIfExists' flag is passed.`
+      );
+      return;
+    }
+
+    throw new RuntimeExistsError(
+      `Directory '${destinationPath}' already exists. To force runtime downloading use 'force' flag.`
+    );
+  }
+
+  const runtimeArchiveUrl = getRuntimeArchiveUrl(osInfo);
+  const tmpDir = `temp-ov-runtime-archive-${new Date().getTime()}`;
+  const tempDirectoryPath = path.join(os.tmpdir(), tmpDir);
+
+  try {
+    const filename = path.basename(runtimeArchiveUrl);
+    const archiveFilePath = path.resolve(tempDirectoryPath, filename);
+
+    await fs.mkdir(tempDirectoryPath);
+
+    console.log('Downloading OpenVINO runtime archive...');
+    await downloadFile(runtimeArchiveUrl, filename, tempDirectoryPath, config.proxy);
+    console.log('OpenVINO runtime archive downloaded.');
+
+    await removeDirectory(destinationPath);
+
+    console.log('Extracting archive...');
+    await unarchive(archiveFilePath, destinationPath);
+
+    console.log('The archive was successfully extracted.');
+  } catch (error) {    
+    console.error(`Failed to download OpenVINO runtime: ${error}.`);
+    throw error;
+  } finally {
+    await removeDirectory(tempDirectoryPath);
+  }
+}
+
+/**
+ * The OS information object.
+ * @typedef {Object} OsInfo
+ * @property {NodeJS.Platform} platform
+ * @property {string} arch
+ */
+
+/**
+ * Get information about OS.
+ *
+ * @async
+ * @function getOsInfo
+ * @returns {Promise<OsInfo>}
+ */
+async function getOsInfo() {
+  const platform = os.platform();
+
+  if (!['win32', 'linux', 'darwin'].includes(platform)) {
+    throw new Error(`Platform '${platform}' is not supported.`);
+  }
+
+  const arch = os.arch();
+
+  if (!['arm64', 'armhf', 'x64'].includes(arch)) {
+    throw new Error(`Architecture '${arch}' is not supported.`);
+  }
+
+  if (platform === 'win32' && arch !== 'x64') {
+    throw new Error(`Version for windows and '${arch}' is not supported.`);
+  }
+
+  return { platform, arch };
+}
+
+/**
+ * Check if directory exists.
+ *
+ * @async
+ * @function checkIfDirectoryExists
+ * @param {string} directoryPath - The directory path.
+ * @returns {Promise<boolean>}
+ */
+async function checkIfDirectoryExists(directoryPath) {
+  try {
+    await fs.access(directoryPath);
+    return true;
+  } catch (error) {
+    if (error.code === codeENOENT) {
+      return false;
+    }
+    throw error;
+  }
+}
+
+/**
+ * Get OpenVINO runtime archive URL.
+ *
+ * @function getRuntimeArchiveUrl
+ * @param {OsInfo} osInfo - The OS related data.
+ * @returns {string}
+ */
+function getRuntimeArchiveUrl(osInfo) {
+  const packageNameTemplate = packageJson.binary['package_name'];
+
+  const packageName = packageNameTemplate
+    .replace('{platform}', osInfo.platform)
+    .replace('{arch}', osInfo.arch)
+
+  const { host, remote_path } = packageJson.binary;
+
+  return new URL(path.join(remote_path, packageName), host).toString();
```

Comment:
Will `new URL` replace `\` with `/` from `path.join` in case of using win machine?

Reply:
URL constructor output doesn't have difference between OS's, but I replaced